Server/controllers/monitorController.js

```diff
@@ -82,4 +85,68 @@ const getMonitorsByUserId = async (req, res) => {
   }
 };
 
-module.exports = { getAllMonitors, getMonitorById, getMonitorsByUserId };
+/**
+ * Creates a new monitor
+ * @async
+ * @param {Express.Request} req
+ * @param {Express.Response} res
+ * @returns {Promise<Express.Response>}
+ * @throws {Error}
+ */
+
+const createMonitor = async (req, res) => {
+  const { error } = createMonitorValidation.validate(req.body);
+  if (error) {
+    return res
+      .status(400)
+      .json({ success: false, msg: error.details[0].message });
+  }
+
+  try {
+    const monitor = await req.db.createMonitor(req, res);
+    return res
+      .status(200)
+      .json({ success: true, msg: "Monitor created", data: monitor });
+  } catch (error) {
+    logger.error(error.message, { service: "monitor" });
+    return res.status(500).json({ success: false, msg: error.message });
+  }
+};
+
+// Delete Monitor
+/**
+ * Delete a monitor by ID
+ * @async
+ * @param {Express.Request} req
+ * @param {Express.Response} res
+ * @returns {Promise<Express.Response>}
+ * @throws {Error}
+ */
+
+const deleteMonitor = async (req, res) => {
+  const { error } = getMonitorByIdValidation.validate(req.params);
+  if (error) {
+    return res
+      .status(400)
+      .json({ success: false, msg: error.details[0].message });
+  }
+  try {
+    const monitor = await req.db.deleteMonitor(req, res);
+    return res
+      .status(200)
+      .json({ success: true, msg: "Monitor deleted", data: monitor });
```

Comment:
A couple of things worth discussing;
1. Should we return 204 instead of 200 as a status code? 
2. Should we delete the related checks and alerts after deleting the monitor inside this controller?
3. More importantly, should we force users to only delete inactive monitors (they need to pause the service first)?

Reply:
Good points, my thoughts:

1.  Very good point, I should update the various status codes to be more semantically accurate.  200 is semantically correct but not very specific.  I'll update codes to provide more information.

2.  Yes, I don't think it makes sense to keep the checks and alerts for a monitor around once the monitor has been deleted.  I'll add comments to remind us to do that once we've implemented checks and alerts

3.  I think this can be hidden from the user and handled on the backend.  Their intention when deleting a monitor is to turn it off anyways, so we can handle stopping the service for them, otherwise it's an extra step on their end.

I'll handle 1 and 2 and update the PR, for 3 I'll add some comments to remind us to deal with it once we've got the service hammered out :+1:

Reply:
Awesome! 👍